### DISCUSSION TITLE
SF Symbol Animations

### DISCUSSION THREAD

**Discussion by in03** [swiftDialog/swiftDialog#459] [Ideas]
I love that SwiftDialog supports SF symbols. Any plans to support their native animations? Maybe using similar subcommand syntax as [`listitem`](https://arc.net/l/quote/lcnalyks) commands?

```
- listitem: add, title: <text>[, status: <status>, statustext: <text>]
- listitem: delete, title: <text>
```

Maybe
```
- icon: SF=wifi.exclamationmark, animation: bounce               # Repeat continuously
- icon: SF=wifi.exclamationmark, animation: bounce, delay: 5s    # Repeat with delay

- icon: SF=wifi                                                  # Set enabled icon
- icon: SF=wifi.slash, animation: replace                        # Animate disabled slash
```

I feel these would be great for dialogs using command files being used to check conditions.
The magic replace animations are particularly cool.
Obviously for aesthetic reasons.

**Comment by bartreardon** [comment]
you can already to a limited extent

https://github.com/swiftDialog/swiftDialog/wiki/Customising-the-Icon#animated-sf-symbols

> `--icon sf=rainbow,colour=auto,animation=variable`
> 
> Limited support at this time using the following animation types:
> 
>  - "variable":
>  - "variable.reversing"
>  - "variable.iterative"
>  - "variable.iterative.reversing"
>  - "variable.cumulative"
>  - "pulse"
>  - "pulse.bylayer"

**Reply by in03** [reply]
Sorry! I have no idea how I missed this. Apologies for my ignorance.

**Reply by bartreardon** [reply]
all good, no need to apologise 🙂

There's a lot of docco now and probably not the best organised.
